akka-docs/rst/java/code/docs/http/javadsl/server/directives/RouteDirectivesExamplesTest.java

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright (C) 2016-2016 Lightbend Inc. <http://www.lightbend.com>
+ */
+package docs.http.javadsl.server.directives;
+
+import akka.http.javadsl.model.HttpEntities;
+import akka.http.javadsl.model.HttpRequest;
+import akka.http.javadsl.model.Uri;
+import akka.http.javadsl.model.headers.ContentType;
+import akka.http.javadsl.model.ContentTypes;
+import akka.http.javadsl.model.HttpResponse;
+import akka.http.javadsl.model.StatusCodes;
+import akka.http.javadsl.server.Rejections;
+import akka.http.javadsl.server.Route;
+import akka.http.javadsl.testkit.JUnitRouteTest;
+import org.junit.Test;
+
+import java.util.Collections;
+
+/**
+ * @author Hawstein
+ */
+public class RouteDirectivesExamplesTest extends JUnitRouteTest {
+
+  @Test
+  public void testComplete() {
+    //#complete
+    final Route route = route(
+      path("a", () -> complete(HttpResponse.create().withEntity("foo"))),
+      path("b", () -> complete(StatusCodes.OK)),
+      path("c", () -> complete(StatusCodes.CREATED, "bar")),
+      path("d", () -> complete(StatusCodes.get(201), "bar")),
+      path("e", () ->
+        complete(StatusCodes.CREATED,
+                 Collections.singletonList(ContentType.create(ContentTypes.TEXT_PLAIN_UTF8)),
+                 HttpEntities.create("bar"))),
+      path("f", () ->
+        complete(StatusCodes.get(201),
+                 Collections.singletonList(ContentType.create(ContentTypes.TEXT_PLAIN_UTF8)),
+                 HttpEntities.create("bar"))),
+      path("g", () -> complete("baz"))
+    );
+
+    // tests:
+    testRoute(route).run(HttpRequest.GET("/a"))
+      .assertStatusCode(StatusCodes.OK)
+      .assertEntity("foo");
+
+    testRoute(route).run(HttpRequest.GET("/b"))
+      .assertStatusCode(StatusCodes.OK)
+      .assertEntity("OK");
```

Comment:
Which is weird because the test passed... I'll see if we're confident they actually run hm